### ISSUE TITLE
appcast embedded release notes doesn't seem to work if release notes are too long in Sparkle v2

### ISSUE THREAD

**Issue by godbout** [sparkle-project/Sparkle#2039]
## Summary

generating appcasts through the `generate_appcast` command has always worked well and embedded the release notes found in the `html` file that has the same name as the archive. but it seems that if the release notes are too long, then they are not embedded. that definitely may be the normal behavior, and a good thing. but i can't find any info about this in the doc. someone could confirm that this is the expected behavior?
(in my case it points instead to the `html` file on my server, which fails because of pretty URLs.)

## Possible Fix

N/A

## Version

2.x master

**Comment by godbout** [comment]
e.g.

this works:
`
<h2>improve</h2>
<ul>
    <li>performance tweaks in Operator Modes</li>
    <li>improve Keyboard Strategy Normal Mode cc for Non Text Elements</li>
    <li>improve Keyboard Strategy Normal Mode and Visual Mode for V, v for Non Text Elements</li>
    <li>improve Keyboard Strategy Normal Mode x for Non Text Elements</li>
    <li>improve Keyboard Strategy Visual Mode c for Non Text Elements</li>
    <li>improve Keyboard Strategy Visual Mode d for Non Text Elements</li>
</ul>
<h2>new</h2>
<ul>
    <li>add support for Keyboard Strategy Normal Mode cG, dG for Non Text Elements</li>
    <li>add support for Keyboard Strategy Normal Mode cgg, dgg for Non Text Elements</li>
    <li>add support for Keyboard Strategy Visual Mode G, gg for Non Text Elements</li>
</ul>
<h2>summary</h2>
for example if you're in Mail, now you can enter kindaVim Normal Mode (escape by default) and:
<ul>
    <li>delete a mail with x</li>
</ul>
`

but this doesn't (one more <li>):
`
<h2>improve</h2>
<ul>
    <li>performance tweaks in Operator Modes</li>
    <li>improve Keyboard Strategy Normal Mode cc for Non Text Elements</li>
    <li>improve Keyboard Strategy Normal Mode and Visual Mode for V, v for Non Text Elements</li>
    <li>improve Keyboard Strategy Normal Mode x for Non Text Elements</li>
    <li>improve Keyboard Strategy Visual Mode c for Non Text Elements</li>
    <li>improve Keyboard Strategy Visual Mode d for Non Text Elements</li>
</ul>
<h2>new</h2>
<ul>
    <li>add support for Keyboard Strategy Normal Mode cG, dG for Non Text Elements</li>
    <li>add support for Keyboard Strategy Normal Mode cgg, dgg for Non Text Elements</li>
    <li>add support for Keyboard Strategy Visual Mode G, gg for Non Text Elements</li>
</ul>
<h2>summary</h2>
for example if you're in Mail, now you can enter kindaVim Normal Mode (escape by default) and:
<ul>
    <li>delete a mail with x</li>
    <li>delete a mail and go back to Insert Mode with cc</li>
</ul>
`

trying many different combinations and it does seem to be about release notes length.

**Comment by godbout** [comment]
the usual `find answer have posting`. i guess that's it: https://github.com/sparkle-project/Sparkle/blob/a81f8f3d2d042cd5b2e978b6708f195423ebbfb7/generate_appcast/ArchiveItem.swift#L9

**Comment by zorgiepoo** [comment]
Yeah, this has always been the case and is documented in the generate_appcast help page. I'm not too sure if we want to change this. (I wasn't the one that put that restriction)

**Comment by godbout** [comment]
> Yeah, this has always been the case and is documented in the generate_appcast help page for 2.x.
my bad, sorry. had checked on the Sparkle site. i see it now in the cli help yeah, thanks.

no worries for the change. in that case, is there an option or config to decide where the release note link is pointing? in my case it's pointing to `.../kindaVim.html` i guess because that's the name of the file, but on my site it's `.../kindaVim/index.html` (index.html not necessary). coz static site etc etc. i can get around this but if there's an option in Sparkle, even better! thanks.

**Comment by zorgiepoo** [comment]
There is a `release-notes-url-prefix` option but it may not do exactly what you want.

I think I would vote for removing the `1000` limit, though.

**Comment by godbout** [comment]
will 

> There is a `release-notes-url-prefix` option but it may not do exactly what you want.

thanks. will check and investigate.

> I think I would vote for removing the `1000` limit, though.

up to you guys. actually i think the limit makes sense. so that you don't end up with a gigornous appcast.xml.

**Comment by zorgiepoo** [comment]
A hidden command line option for now has landed in [2.1.0-beta.1](https://github.com/sparkle-project/Sparkle/releases/tag/2.1.0-beta.1) in case any other developer runs into this and wants a workaround.

**Comment by godbout** [comment]
thank you!
